src/resource/kv-store.c

```diff
@@ -477,6 +477,49 @@ void *kv_store_iter_current(kv_store_iter_t *iter, size_t *size, kv_store_value_
 	return _get_data(value);
 }
 
+int kv_store_iter_current_size(kv_store_iter_t *iter,
+                               size_t *         int_size,
+                               size_t *         int_data_size,
+                               size_t *         ext_size,
+                               size_t *         ext_data_size)
+{
+	size_t                 iov_size, data_size;
+	struct kv_store_value *value;
+
+	if (!iter || !int_size || !int_data_size || !ext_size || !ext_data_size)
+		return -1;
+
+	if (!(value = iter->current ? hash_get_data(iter->store->ht, iter->current, NULL) : NULL))
+		return -1;
+
+	if (value->ext_flags && KV_STORE_VALUE_VECTOR) {
+		int           i;
+		struct iovec *iov = _get_ptr(value->data);
```

Comment:
Oops. My bad. I'll fix that. But this makes me wonder, is
`typedef enum
{
        KV_STORE_VALUE_INT_ALLOC = UINT32_C(0x00000001),
} kv_store_value_int_flags_t;`
guaranteed to be 32 bits? I thought enum size was compiler dependent. If it's not guaranteed to be at least 32 bits, 
`struct kv_store_value` needs to go back to explicitly declaring the size of int_flags and ext_flags. Otherwise, value->data might not be aligned for storing iov structures.

Reply:
I admit the enums may be a bit unclear area. I've just checked with the C standard and it only says:

  - *The expression that defines the value of an enumeration constant shall be an integer constant expression that has a value representable as an int.*
 
  - *Each enumerated type shall be compatible with char, a signed integer type, or an unsigned integer type. The choice of type is implementation-defined, but shall be capable of representing the values of all the members of the numeration.*

From that, the only thing that standard assures is that the members are compatible with `int`, and that one is 16 bits at least by standard. From that, I understand that if I wanted the members of enum to be of `long int` or `long long int` type, that is completely in the hands of the compiler if it decides to support that or not.

So yes, I think if we wanted to be sure, we should replace all those enums where members are defined as `UINT32_C`,  `UINT64_C` types which means we expect 32 or 64 bit space to be available for these (we will lose nice representation of those flags in debugger then, but I can probably live without it).

GCC and clang seem to use the `int` as the shortest enum member representation and then they extend this as needed so if there's a member with value which is higher than max int, it automatically uses larger type. But this is all compiler's will, nothing defined by the standard!

Reply:
(opened #95 to track the standard/compiler extension/enum and related)